test/sys/test_wait.rs

```diff
@@ -37,6 +37,19 @@ fn test_wait_exit() {
     }
 }
 
+#[test]
+fn test_waitstatus_pid() {
+    let _m = ::FORK_MTX.lock().expect("Mutex got poisoned by another test");
+
+    match fork().unwrap() {
+        Child => {},
```

Comment:
Don't allow the child to return.  It's vulnerable to deadlocks in the test harness code.  Instead, call `_exit(0)`.